src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -962,7 +962,10 @@ private async Task StartRequestAsync(WinHttpRequestState state)
                 // Since the headers have been read, set the "receive" timeout to be based on each read
                 // call of the response body data. WINHTTP_OPTION_RECEIVE_TIMEOUT sets a timeout on each
                 // lower layer winsock read.
-                uint optionData = unchecked((uint)_receiveDataTimeout.TotalMilliseconds);
+                // Timeout.InfiniteTimeSpan will be converted to uint.MaxValue milliseconds (~ 50 days).
+                // The result a of double->uint cast is unspecified and may differ on ARM.
+                // To handle Timeout.InfiniteTimespan correctly, we need to cast to Int32 first.
+                uint optionData = unchecked((uint)(int)_receiveDataTimeout.TotalMilliseconds);
```

Comment:
`unchecked` is unnecessary. We build everything as `unchecked` and have no plans to change it.